pep-0688.rst

```diff
@@ -120,57 +189,58 @@ subclassed at runtime, but supports the ``__instancecheck__`` and
    >>> issubclass(str, Buffer)
    False
 
-The new class can also be used in type annotations:
+Example
+-------
 
-.. code-block:: python
+The following is an example of a Python class that implements the
+buffer protocol:
 
-   def need_buffer(b: Buffer) -> memoryview:
-       return memoryview(b)
-
-   need_buffer(b"xy")  # ok
-   need_buffer("xy")  # rejected by static type checkers
+.. code-block:: python
 
-Usage in stub files
--------------------
+   class MyBuffer:
+       def __init__(self, data: bytes):
+           self.data = bytearray(data)
+           self.view = memoryview(self.data)
+           self.held = False
 
-For static typing purposes, types defined in C extensions usually
-require stub files, as :pep:`described in PEP 484 <484#stub-files>`.
-In stub files, ``types.Buffer`` may be used as a base class to
-indicate that a class implements the buffer protocol.
+       def __buffer__(self, flags: int) -> memoryview:
+           if flags != inspect.BufferFlags.FULL_RO:
+               raise TypeError("Only BufferFlags.FULL_RO supported")
+           if self.held:
+               raise RuntimeError("Buffer already held")
+           self.held = True
+           return self.view
 
-For example, ``memoryview`` may be declared as follows in a stub:
+       def __release_buffer__(self, view: memoryview) -> None:
+           assert self.view is view
+           self.held = False
 
-.. code-block:: python
+   buffer = MyBuffer(b"capybara")
+   with memoryview(buffer) as view:
+       view[0] = ord("C")
 
-   class memoryview(types.Buffer, Sized, Sequence[int]):
-       ...
+   with memoryview(buffer) as view:
+       assert view.tobytes() == b"Capybara"
 
-The ``types.Buffer`` class does not require any special treatment
-by type checkers.
 
 Equivalent for older Python versions
 ------------------------------------
 
 New typing features are usually backported to older Python versions
 in the `typing_extensions <https://pypi.org/project/typing-extensions/>`_
 package. Because the buffer protocol
-is accessible only in C, ``types.Buffer`` cannot be implemented
+is currently accessible only in C, this PEP cannot be fully implemented
 in a pure-Python package like ``typing_extensions``. As a temporary
-workaround, a ``typing_extensions.Buffer``
-`abstract base class <Buffer ABC_>`__ will be provided for Python versions
-that do not have ``types.Buffer`` available.
+workaround, two abstract base classes, ``typing_extensions.Buffer``
+and ``MutableBuffer``, will be provided for Python versions
+that do not have ``collections.abc.Buffer`` available.
 
-For the benefit of
-static type checkers, ``typing_extensions.Buffer`` can be used as
-a base class in stubs to mark types as supporting the buffer protocol.
+For the benefit of static type checkers, ``typing_extensions.Buffer``
+can be used as a base class in stubs to mark types as supporting the buffer protocol.
```

Comment:
Yes, because the protocols won't apply before 3.12 where there's no `__buffer__` method.

Reply:
Ahh, I see. Could you make this more explicit? Something like `In Python versions where objects supporting the buffer protocol have a __buffer__ method, inheriting from collections.abc.Buffer will not be necessary to indicate that an object supports the buffer protocol. However, in older Python versions, it will be necessary to explicitly inherit from typing_extensions.Buffer to indicate to type checkers that a class supports the buffer protocol, since objects supporting the buffer protocol will not have a __buffer__ method.`

Reply:
Thanks, pushed a variation of that wording